frontend/src/metabase/nav/components/SearchBar.unit.spec.tsx

```diff
@@ -0,0 +1,182 @@
+        expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument(),
+      );
+
+      expect(
+        within(screen.getByTestId("search-bar-results-container")).getByText(
+          "Didn't find anything",
+        ),
+      ).toBeInTheDocument();
+    });
+  });
+  describe("focusing on search bar", () => {
+    it("should render `Recent Searches` list when clicking the search bar", async () => {
+      setup();
+      getSearchBar().click();
+      expect(await screen.findByText("Recents ABC")).toBeInTheDocument();
+    });
+
+    it("should render `Nothing here` and a folder icon if there are no recently viewed items", async () => {
+      setup({ recentViewsItems: [] });
+      getSearchBar().click();
+      expect(await screen.findByText("Nothing here")).toBeInTheDocument();
+    });
+  });
+  describe("keyboard navigation", () => {
+    it("should focus on the filter bar when the user tabs from the search bar", () => {
+      setup();
+      getSearchBar().click();
+      userEvent.tab();
+      expect(screen.getByTestId("search-bar-filter-button")).toHaveFocus();
+    });
+    it("should allow navigation through the search results with the keyboard", async () => {
+      setup();
+      getSearchBar().click();
+      userEvent.type(getSearchBar(), "BC");
+
+      await waitFor(() =>
+        expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument(),
+      );
+
+      const resultItems = await screen.findAllByTestId("search-result-item");
+      expect(resultItems.length).toBe(2);
+
+      // tab over the filter button
+      userEvent.tab();
+
+      // There are two search results, each with a link to `Our analytics`,
+      // so we want to navigate to the search result, then the collection link.
+      for (const substring of ["Card ABC", "Card BCD"]) {
+        userEvent.tab();
+
+        const filteredElement = resultItems.find(element =>
+          element.textContent?.includes(substring),
+        );
+
+        expect(filteredElement).not.toBeUndefined();
+        expect(filteredElement).toHaveFocus();
+
+        userEvent.tab();
+
+        expect(
+          within(filteredElement as HTMLElement).getByText("Our analytics"),
+        ).toHaveFocus();
+      }
+    });
+  });
+  describe("populating existing query", () => {
+    it("should populate text and highlight filter button when a query is in the search bar", () => {
+      setup({
+        initialRoute: "/search?q=foo&type=card",
+      });
+
+      expect(getSearchBar()).toHaveValue("foo");
+
+      expect(
+        screen.getByTestId("highlighted-search-bar-filter-button"),
+      ).toBeInTheDocument();
+    });
+
+    it("should not populate text or highlight filter button on non-search pages", () => {
+      setup({
+        initialRoute: "/collection/root?q=foo&type=card&type=dashboard",
+      });
+
+      expect(getSearchBar()).toHaveValue("");
+
+      expect(
+        screen.getByTestId("search-bar-filter-button"),
```

Comment:
Also, it's not apparent why we assert `test-id="search-bar-filter-button"` here and `test-id="highlighted-search-bar-filter-button"` in the previous test, when the only differences I can spot are:
1. The path are different `/search` vs `/collection/root`
2. The search param is different, this one has `type=dashboard` and the rest are the same as the previous test's search params.

Reply:
We want to make sure that we don't have random search filters applied in the search bar when a user navigates to a different page (and would cause the filter button to be highlighted in the search bar), so we don't want the search bar "intercepting" query parameters from a non-search URL. It goes with the whole "conditional data-testid" from above that we probably shouldn't be using anyway